File: LeetCode/563. Binary Tree Tilt.cpp
/***  KNOWLEDGE IS POWER  ***/

class Solution {
public:
    int ans = 0;
    int findTilt(TreeNode* root) {
        dfs(root);
        return ans ;
    }
    int dfs(TreeNode* root) {
        if(!root) 
            return 0 ;
        int left = dfs(root -> left);
        int right = dfs(root -> right) ;
        ans += abs(left - right);       
        return left + right + root->val ;
    }
};

File: LeetCode/563. Binary Tree Tilt.py
###  KNOWLEDGE IS POWER  ###

class Solution(object):
    def findTilt(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        self.tilt_sum = 0
        def dfs(root):
            if not root: return 0
            left_sum, right_sum = dfs(root.left), dfs(root.right)
            self.tilt_sum += abs(left_sum - right_sum)
            return left_sum + right_sum + root.val
        dfs(root)
        return self.tilt_sum
